src/app/features/location/address-edit/address-edit.component.html

```diff
@@ -65,7 +65,21 @@
 </div>
 
 @if (!disabled) {
-  <app-address-search
-    (locationSelected)="updateFromAddressSearch($event)"
-  ></app-address-search>
+  <div class="margin-bottom-regular container">
+    <div class="flex-row gap-regular align-start full-width">
+      <app-address-search
+        (locationSelected)="updateFromAddressSearch($event)"
+        class="full-width"
+      ></app-address-search>
+      <button
+        mat-icon-button
+        (click)="updateLocationFromGps()"
+        matTooltip="To accurately log your current address, please allow GPS access."
+        i18n-matTooltip
+      >
+        <fa-icon *ngIf="!gpsLoading" icon="location-crosshairs"></fa-icon>
+        <mat-spinner *ngIf="gpsLoading" diameter="24"></mat-spinner>
+      </button>
```

Comment:
Should we create a component for this? Let's check what patterns we are following